src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -223,6 +224,20 @@ private bool MatchString(string str)
             {
                 if (runtextend - runtextpos < c)
                 {
+                    // If MatchString was called after a greedy op such as a .*, we would have zipped runtextpos to the end without really examining any characters. Reset to maxBacktrackPos here as an optimization
+                    if (_maxBacktrackPosition != -1 && runtextpos > _maxBacktrackPosition)
+                    {
+                        // If lastIndexOf is -1, we backtrack to the max extent possible.
+                        runtextpos = _maxBacktrackPosition;
+                        ReadOnlySpan<char> runtextSpan = runtext.AsSpan(_maxBacktrackPosition);
+                        int lastIndexOf = runtextSpan.LastIndexOf(str);
```

Comment:
This is doing an ordinal comparison for the whole string.  Doesn't this need to factor in the culture (at which point, given how the rest of the lib handles culture, it would need to do a search for just the single character)?  And, what about casing... isn't this only relevant if we're not in an ignore case region?  If I'm right that this is an issue, it also means we're missing tests.

Reply:
> And, what about casing... isn't this only relevant if we're not in an ignore case region?

Yup. Nice catch. Fixed now.

> If I'm right that this is an issue, it also means we're missing tests.

I don't think so. In this PR's case, we didn't see any failing tests because `lastIndexOf` would return -1 and we wouldn't change `runtextpos`. So effectively no changes were made to the interpreter. I also see many tests for `IgnoreCase`. Here's one: `yield return new object[] { "abc", "aBc", RegexOptions.IgnoreCase, 0, 3, true, "aBc" };`

> Doesn't this need to factor in the culture (at which point, given how the rest of the lib handles culture, it would need to do a search for just the single character)?

This question I'm not very sure about. Further down in this method (line 256), we seem to only be checking for case sensitivity. I don't see anything related to culture. Maybe there's an existing bug here? What do you think about me opening an issue to investigate this? We can merge this PR in the meanwhile and if I find an issue with culture, I can fix both places (this optimization and the if conditions on line 256).

Reply:
>> If I'm right that this is an issue, it also means we're missing tests.

> I don't think so.

There was a functional bug in the implementation, yet all tests were passing... that typically means there's a missing test.  For example, without the fix you made to add the case insensitive check, this test would likely have failed:
```C#
yield return new object[] { @"abc.*def", "abcghiDEF", RegexOptions.IgnoreCase, 0, 9, true, "abcghiDEF" };
```
Try reverting your fix and adding that to Match_Basic_TestData.  I expect everything will pass when this is run on netfx but, with this PR without the casing fix, this would fail on core.

>  I also see many tests for IgnoreCase. Here's one

Sure, but that doesn't involve `.*` or searching for the text after it.  Even with high code coverage, we likely have many test holes in regex because there are simply so many combinations of how different features interact with each other.

>> Doesn't this need to factor in the culture

> This question I'm not very sure about. 

Ignore this one... I was thinking of a different situation.

Reply:
> There was a functional bug in the implementation, yet all tests were passing... that typically means there's a missing test
> @"abc.*def", "abcghiDEF",

TLDR: Yup, you are right. I added this as a unit test.

Before my first reply, I tested locally with `ab.*foo` and input `abFoo`. Surprisingly this passed because the backtracking happened to match `foo` with `Foo`, so I didn't investigate it further. It will of course fail to match `abxFoo`, but that was hidden by my simple test. My bad